drivers/power/lt3074/iio_lt3074.c

```diff
@@ -0,0 +1,727 @@
+				  intptr_t priv)
+{
+	struct lt3074_iio_desc *iio_lt3074 = dev;
+	struct lt3074_dev *lt3074 = iio_lt3074->lt3074_dev;
+	int ret, word;
+	int32_t vals[2];
+
+	if (!dev)
+		return -EINVAL;
+
+	if (!iio_lt3074->lt3074_dev)
+		return -EINVAL;
+
+	ret = lt3074_read_word_data(lt3074, priv, &word);
+	if (ret)
+		return ret;
+
+	vals[0] = (int32_t)(word / MILLI);
+	vals[1] = (int32_t)((word * MILLI) % MICRO);
+
+	return iio_format_value(buf, len, IIO_VAL_INT_PLUS_MICRO, 2, vals);
+}
+
+/**
+ * @brief Handles the write request for limits global attributes.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return ret    - Result of the reading procedure.
+ * 		    In case of success, the size of the read data is returned.
+*/
+static int lt3074_iio_write_limits(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel,
+				   intptr_t priv)
+{
+	struct lt3074_iio_desc *iio_lt3074 = dev;
+	struct lt3074_dev *lt3074 = iio_lt3074->lt3074_dev;
+	int limit;
+	int32_t val1, val2;
+
+	iio_parse_value(buf, IIO_VAL_INT_PLUS_MICRO, &val1, &val2);
+
+	limit = (int)(val1 * MILLI + val2 / MILLI);
+
+	return lt3074_set_limit(lt3074, (enum lt3074_limit_type)priv, limit);
+}
+
+/**
+ * @brief Initializes the LT3074 IIO descriptor.
+ * @param iio_desc - The iio device descriptor.
+ * @param init_param - The structure that contains the device initial parameters.
+ * @return 0 in case of success, an error code otherwise.
+ */
+int lt3074_iio_init(struct lt3074_iio_desc **iio_desc,
+		    struct lt3074_iio_desc_init_param *init_param)
+{
+	struct lt3074_iio_desc *descriptor;
+	int ret;
+
+	if (!init_param || !init_param->lt3074_init_param)
+		return -EINVAL;
+
+	descriptor = no_os_calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	ret = lt3074_init(&descriptor->lt3074_dev,
+			  init_param->lt3074_init_param);
+	if (ret)
+		goto dev_err;
+
+	descriptor->iio_dev = &lt3074_iio_dev;
+
+	*iio_desc = descriptor;
+
+	return 0;
+
+dev_err:
+	lt3074_iio_remove(descriptor);
+
+	return ret;
+}
+
+/**
+ * @brief Free resources allocated by the init function.
+ * @param iio_desc - The iio device descriptor.
+ * @return 0 in case of success, an error code otherwise.
+ */
+int lt3074_iio_remove(struct lt3074_iio_desc *iio_desc)
+{
+	if (!iio_desc)
+		return -ENODEV;
+
+	no_os_free(iio_desc->iio_dev->channels);
+	lt3074_remove(iio_desc->lt3074_dev);
+	no_os_free(iio_desc);
+
+	return 0;
+}
+
+static struct iio_attribute lt3074_chan_attrs[] = {
+	{
+		.name = "raw",
+		.show = lt3074_iio_read_raw,
+	},
+	{
+		.name = "scale",
+		.show = lt3074_iio_read_scale,
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+static struct iio_attribute lt3074_global_attrs[] = {
+	{
+		.name = "vout_margin_low",
+		.show = lt3074_iio_read_vout_margin,
```

Comment:
please use a single function for all of these if they are similar enough to avoid code duplication